apertium/stream.cc

```diff
@@ -645,56 +645,56 @@ void Stream::outputLexicalUnit(
          // Call .end() each iteration to save memory.
          other_analysis != lexical_unit.TheAnalyses.end(); ++other_analysis) {
       if (*other_analysis != *analysis)
-        output << L"/" << *other_analysis;
+        output << "/" << *other_analysis;
     }
   }
 
-  output << L"$";
+  output << "$";
 }
 
-Stream::PreviousCaseType::PreviousCaseType(const wchar_t &PreviousCase_)
+Stream::PreviousCaseType::PreviousCaseType(const UChar &PreviousCase_)
     : ThePreviousCase(PreviousCase_), isPreviousCharacter(true) {}
 
 bool Stream::is_eof_throw_if_not_TheCharacterStream_good() const {
   if (TheCharacterStream.eof())
     return true;
 
   if (!TheCharacterStream) {
-    std::wcerr << L"State bad " << TheCharacterStream.good() << " "
+    std::cerr << "State bad " << TheCharacterStream.good() << " "
                                 << TheCharacterStream.eof() << " "
                                 << TheCharacterStream.fail() << " "
                                 << TheCharacterStream.bad() << "\n";
-    std::wstringstream Message;
-    Message << L"can't get const wchar_t: TheCharacterStream not good";
+    std::stringstream Message;
+    Message << "can't get const UChar: TheCharacterStream not good";
     throw Exception::Stream::TheCharacterStream_not_good(
         Message_what(Message));
   }
 
   return false;
 }
 
-std::wstring Stream::Message_what(const std::wstringstream &Message) const {
-  std::wstringstream what_;
+UString Stream::Message_what(const std::stringstream &Message) const {
+  std::stringstream what_;
 
   if (TheFilename)
-    what_ << std::wstring(TheFilename->begin(), TheFilename->end()) << L": ";
+    what_ << UString(TheFilename->begin(), TheFilename->end()) << ": ";
 
-  what_ << TheLineNumber << L":" << TheLine.size() << L": " << Message.str()
-        << L'\n' << TheLine << L'\n' << std::wstring(TheLine.size() - 1, L' ')
-        << L'^';
-  return what_.str();
+  what_ << TheLineNumber << ":" << TheLine.size() << ": " << Message.str()
+        << '\n' << TheLine << '\n' << UString(TheLine.size() - 1, ' ')
+        << '^';
+  return to_ustring(what_.str().c_str());
```

Comment:
Why `.c_str()` ?

Reply:
because I had only defined `to_ustring(const char*)` - I suppose this happens often enough that it would make sense to also have `to_ustring(std::string)`

Reply:
Oh, I also misread it as the .c_str() being outside the (). The cheap catch-all for char* and string is string_view.